code/client/munkilib/updatecheck.py

```diff
@@ -2605,7 +2650,7 @@ def getmanifest(partialurl, suppress_errors=False):
         raise ManifestException(errormsg)
     else:
         # plist is valid
-        MANIFESTS[manifestname] = manifestpath
+        addToManifestCache(manifestname, manifestpath)
         return manifestpath
 
 
```

Comment:
Probably could now also use a new cleanUpManifests() function, similar to cleanUpCatalogs, to remove any manifests from the local cache that are no longer used, including any empty subdirectories. (Be sure to leave the SelfServeManifest alone if it exists!)